contracts/SocialController.sol

```diff
@@ -0,0 +1,246 @@
+            _amount <= MAX_SAFE_INTEGER - creator.numOfOutstandingTokens,
+            "You need to purchase a smaller amount of tokens."
+        );
+        require(creator.isCreator, "This token ID is not valid.");
+
+        CreatorPricingFactors memory pricingFactors = creator.pricingFactors;
+        uint256 multipleFactor = pricingFactors.multipleFactor;
+        uint256 growthFactor = pricingFactors.growthFactor;
+        uint256 constantFactor = pricingFactors.constantFactor;
+
+        uint256 numOfOutstandingTokens = creator.numOfOutstandingTokens;
+
+        uint256 totalPrice;
+        if (_amount == 1) {
+            totalPrice = _getBuyPrice(multipleFactor, growthFactor, constantFactor, numOfOutstandingTokens);
+        } else {
+            totalPrice = _getBatchBuyPrice(
+                multipleFactor,
+                growthFactor,
+                constantFactor,
+                numOfOutstandingTokens,
+                _amount
+            );
+        }
+
+        IERC20 dai = IERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+        dai.transferFrom(msg.sender, address(this), totalPrice);
+        _mint(msg.sender, _tokenId, _amount, "");
+        creatorById[_tokenId].numOfOutstandingTokens += _amount;
+
+        emit BuyTokenSuccess(_tokenId, _amount, totalPrice);
+    }
+
+    /**
+     * @notice For users to sell specific creator tokens by quantity. Tokens are burned with each sell.
+     * @param _tokenId              The token ID of the creator that the user wishes to sell.
+     * @param _amount               The amount that the user wishes to sell.
+     */
+    function sellToken(uint256 _tokenId, uint256 _amount) external {
+        require(_amount > 0, "You have not input a valid amount for selling.");
+        require(balanceOf(msg.sender, _tokenId) >= _amount, "You do not have enough creator tokens to sell.");
+
+        Creator memory creator = creatorById[_tokenId];
+        uint256 numOfOutstandingTokens = creator.numOfOutstandingTokens;
+        require(numOfOutstandingTokens > 0, "There are no more creator tokens left.");
+
+        CreatorPricingFactors memory pricingFactors = creator.pricingFactors;
+        uint256 multipleFactor = pricingFactors.multipleFactor;
+        uint256 growthFactor = pricingFactors.growthFactor;
+        uint256 constantFactor = pricingFactors.constantFactor;
+
+        uint256 totalPrice;
+        if (_amount == 1) {
+            totalPrice = _getSellPrice(multipleFactor, growthFactor, constantFactor, numOfOutstandingTokens);
+        } else {
+            totalPrice = _getBatchSellPrice(
+                multipleFactor,
+                growthFactor,
+                constantFactor,
+                numOfOutstandingTokens,
+                _amount
+            );
+        }
+
+        _burn(msg.sender, _tokenId, _amount);
+        creatorById[_tokenId].numOfOutstandingTokens -= _amount;
+
+        IERC20 dai = IERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F);
```

Comment:
Here and `LINE199` have duplicated `magic numbers`. Constant variable might be better.

Reply:
Updated to constant